oneandone/src/main/java/org/apache/jclouds/oneandone/rest/compute/OneandoneComputeServiceAdapter.java

```diff
@@ -62,77 +61,95 @@
    protected Logger logger = Logger.NULL;
 
    private final CleanupResources cleanupResources;
+   private final GenerateHardwareRequest generateHardwareRequest;
    private final OneAndOneApi api;
    private final Predicate<Server> waitServerUntilAvailable;
-   private final PasswordGenerator.Config passwordGenerator;
+   private final String baremetalModelsKey = "BMC";
 
    @Inject
    OneandoneComputeServiceAdapter(OneAndOneApi api, CleanupResources cleanupResources,
+           GenerateHardwareRequest generateHardwareRequest,
            @Named(POLL_PREDICATE_SERVER) Predicate<Server> waitServerUntilAvailable,
            PasswordGenerator.Config passwordGenerator) {
```

Comment:
as you have removed it from the constructor, why do you still need to inject it?

Reply:
It is still in the constructor as i need to use it.

Reply:
`PasswordGenerator.Config passwordGenerator` is not used, remove it